tests/test_util.py

```diff
@@ -2083,3 +2085,25 @@ def test_map_to_user_params():
         "arg_none": None,
     }
     assert get_args(user_params) == {"arg1": 1, "arg2": 2}
+
+
+def test_create_tar_gz_archive():
+    """Unittest for create_tar_gz_archive method"""
+
+    test_file = 'foo'
+    content = 'bar'
+    archive_path = create_tar_gz_archive(file_name=test_file, file_content=content)
+
+    with tarfile.open(archive_path) as tar:
+
+        assert len(tar.getnames()) == 1, 'Tar archive does not contain 1 file'
+        assert tar.getnames()[0] == test_file, f'Tar archive does not contain {test_file} file'
+
+        tar.extractall('/tmp')
+        extracted_file = f'/tmp/{test_file}'
```

Comment:
Could this be the `tmpdir` fixture instead of hardcoded /tmp? Unit tests that write directly to my system tmpdir make me a bit nervous :smile:

Reply:
I changed it to tmpdir :D .. also updated commit message. PTAL :)